cmd/mdatagen/main.go

```diff
@@ -66,6 +66,10 @@ func run(ymlPath string) error {
 				filepath.Join(ymlDir, "generated_component_test.go"), md, packageName); err != nil {
 				return err
 			}
+			if err = generateFile(filepath.Join(tmplDir, "package_test.go.tmpl"),
```

Comment:
I'm not very familiar with the inner-workings of `mdatagen`, but would it be very hard to modify so that sub-packages also have `goleak` generated?

For example:
```
.//consumer/consumertest/package_test.go
.//consumer/consumererror/package_test.go
.//consumer/package_test.go
```
_It's likely out of the scope of this PR, just raising as something that could be done here, or will need to be done later._